### PR TITLE
fix: block events and error handling

### PR DESCRIPTION
**Pull request by jpgundrum**
- Fixed the bug in the code where the events are not populated correctly. 
- Increased functionality by calling _transactionError() which creates error logs for the user. 
- Added logic to return the block hash when creating a DID.

### CODE REVIEW

File: packages/sdk/src/modules/did/index.ts [hunk 1]
```diff
@@ -92,15 +92,15 @@ export class Did extends Base {
       );
 
       const nonce = await this._getNonce(keyPair.address);
-      await this._newSignTx({nonce, address: keyPair, extrinsics: attributeExtrinsic});
+      const eventData = await this._newSignTx({nonce, address: keyPair, extrinsics: attributeExtrinsic});
```
Comment (jpgundrum) [inline packages/sdk/src/modules/did/index.ts:95]: Returning back the peaqEvent created in the ./base/index.ts file. The event contains the hash of the block the transaction executed on. 

Any other places in the sdk we should return this block hash?

File: packages/sdk/src/modules/base/index.ts [hunk 2]
```diff
@@ -14,10 +14,20 @@ import { AnyTuple } from '@polkadot/types-codec/types';
 import { truncatedString } from '../../utils';
 
 type TErrorData = {
-  Module?: {
-    index?: string;
-    error?: string;
-  };
+  dispatchError: {
```
Comment (jpgundrum) [inline packages/sdk/src/modules/base/index.ts:17]: Needed to have 'dispatchError' in the object in order to successfully read the error. 'dispatchInfo' contains more information, but not relevant to be logged to the user.

File: packages/sdk/src/modules/base/index.ts [hunk 3]
```diff
@@ -90,17 +100,15 @@ export class Base {
 
   protected async _transactionError(
     method: string,
-    eventData: PeaqEventData[]
-  ): Promise<{ documentation: string[]; name: string } | null> {
+    eventData: PeaqEventData
+  ): Promise<{ documentation: string[]; name: string, section: string } | null> {
```
Comment (jpgundrum) [inline packages/sdk/src/modules/base/index.ts:104]: The added 'section' return string contains more error information which helps the user debug their problem.

File: packages/sdk/src/modules/base/index.ts [hunk 4]
```diff
@@ -213,30 +223,31 @@ export class Base {
                       unsubscribeNewHeads();
                     }
 
-                    const eventsTriggeredByTx: any = []
```
Comment (jpgundrum) [inline packages/sdk/src/modules/base/index.ts:216]: Previous event logic was returning an empty list. It now returns a list of peaqEvents. The code iterates through the events in the current block to build the peaqEvents object. It checks if any event contains an extrinsic failed. If it does, a transactionError is built out by calling the _transactionError() function.

**Review by irediaes — APPROVED** [review]

**Pull request merged**